great_expectations/validation_operators/types/validation_operator_result.py

```diff
@@ -0,0 +1,258 @@
+            self._success = all(
+                [
+                    run_result["validation_result"].success
+                    for run_result in self.run_results.values()
+                ]
+            )
+        return self._success
+
+    @property
+    def validation_results_by_expectation_suite_name(self) -> dict:
+        if self._validation_results_by_expectation_suite_name is None:
+            self._validation_results_by_expectation_suite_name = {
+                expectation_suite_name: [
+                    run_result["validation_result"]
+                    for run_result in self.run_results.values()
+                    if run_result["validation_result"].meta["expectation_suite_name"]
+                    == expectation_suite_name
+                ]
+                for expectation_suite_name in self.expectation_suite_names
+            }
+        return self._validation_results_by_expectation_suite_name
+
+    @property
+    def validation_results_by_data_asset_name(self) -> dict:
+        if self._validation_results_by_data_asset_name is None:
+            validation_results_by_data_asset_name = {}
+            for data_asset_name in self.data_asset_names:
+                if data_asset_name == "__none__":
+                    validation_results_by_data_asset_name[data_asset_name] = [
+                        data_asset["validation_results"]
+                        for data_asset in self.data_assets_validated
+                        if data_asset["batch_kwargs"].get("data_asset_name") is None
+                    ]
+                else:
+                    validation_results_by_data_asset_name[data_asset_name] = [
+                        data_asset["validation_results"]
+                        for data_asset in self.data_assets_validated
+                        if data_asset["batch_kwargs"].get("data_asset_name")
+                        == data_asset_name
+                    ]
+            self._validation_results_by_data_asset_name = (
+                validation_results_by_data_asset_name
+            )
+        return self._validation_results_by_data_asset_name
+
+    @property
+    def statistics(self) -> dict:
+        if self._statistics is None:
+            data_asset_count = len(self.data_assets_validated)
+            validation_result_count = len(self.validation_results)
+            successful_validation_count = len(
+                [
+                    validation_result
+                    for validation_result in self.validation_results
+                    if validation_result.success
+                ]
+            )
+            unsuccessful_validation_count = (
+                validation_result_count - successful_validation_count
+            )
+            successful_validation_percent = (
+                successful_validation_count / validation_result_count
+            ) * 100
+
+            self._statistics = {
+                "data_asset_count": data_asset_count,
```

Comment:
What else would be useful here?

Reply:
Trend lines across batches

Reply:
It might be nice to expose one level deeper of statistics such as the statistics found in each individual validation result. So without much work a user could generate a simple report that shows the number of expectations in each validation that passed.

Just spit-balling here, I could imagine something like:
```python
validation_result.statistics.details_by_suite
{
    "foo.warning": [ValidationResult1_stats],
    "foo.critical": [ValidationResult1_stats, ValidationResult2_stats],
 }
```

The intent is to make it easier to get detailed stats from such a high level aggregate without forcing the user to de-nest them themselves.